pandas/core/algorithms.py

```diff
@@ -1824,16 +1825,42 @@ def diff(arr, n: int, axis: int = 0):
         number of periods
     axis : int
         axis to shift on
+    stacklevel : int
```

Comment:
This is private, so I think OK to accept an internal keyword here.

We need this to get the right stacklevel for the deprecation method since both Series.diff and DataFrame.diff call in here, but with diffferent levels of indirection.